drivers/pwm/pwm_w91.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Telink Semiconductor (Shanghai) Co., Ltd.
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/pwm.h>
+#include <ipc/ipc_based_driver.h>
+
+#define LOG_LEVEL CONFIG_PWM_LOG_LEVEL
+LOG_MODULE_REGISTER(pwm_telink);
+
+/* Driver dts compatibility: telink,w91_pwm */
+#define DT_DRV_COMPAT telink_w91_pwm
+
+#define MAX_CYCLE 1638u
```

Comment:
Is it value from HW? usually this value is 2^n

Reply:
This value isdescribed in "tlsr_timer.h", but there is no define for this:
struct tlsr_timer_cfg_pwm {
	uint16_t high; /* usec, up to ~1638 usec (UINT16_MAX / 40MHz) */
	uint16_t low;  /* usec, up to ~1638 usec (UINT16_MAX / 40MHz) */
	uint8_t park;  /* level of PWM signal when the timer is stopped */
};